superset/charts/api.py

```diff
@@ -503,6 +504,50 @@ def data(self) -> Response:
 
         return response
 
+    @expose("/data/stop", methods=["POST"])
```

Comment:
optional: maybe it would be simpler to set this endpoint to `/data/stop/<id>` HTTP POST or PUT, no need for a payload

Reply:
Curious why if this endpoint takes a `dashboard_id` value, does it live in the `charts` namespace?

Reply:
While the `dashboard` namespace was considered, the main use case for this endpoint is to terminate pending chart data queries, hence the `chart` namespace. The signature will likely be broadened in the future, making it possible to cancel queries based on other parameters (`cache_key`, `slice_id` etc). This endpoint will likely change significantly when the async framework is fully implemented, and will be revisited then.

Reply:
I feel there could be 2 use cases: terminate pending **query(s)** by dashboard_id, or terminate pending **query** by chart id. Currently airbnb has internal API to do 1), but i feel in the future (when the async framework is fully implemented), Superset should be able to do both.